### PR TITLE
[mina-archive-migration package only] opam upgrade caqti

### PR DESCRIPTION
**Pull request by emberian**

### CODE REVIEW

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

File: buildkite/scripts/build-release.sh
Comment (SanabriaRusso) [inline buildkite/scripts/build-release.sh:8]: Is this a good idea 🤔 ?

Comment (emberian) [reply buildkite/scripts/build-release.sh:8]: We can add one more level of hash to pin it to an exact patch.

Could also add the patch here to git.

**Comment by SanabriaRusso** [comment]
`check-graphsql-schema` fails due to an unused variable (https://buildkite.com/o-1-labs-2/mina-o-1-labs/builds/33164#018e5cdc-4043-4c8f-ad4e-7e228ebae69a/53-546). But I think from infra-side this PR is good to go. Any observation @dkijania ?

File: buildkite/scripts/build-release.sh
Comment (SanabriaRusso) [reply buildkite/scripts/build-release.sh:8]: I think adding it here would be clearer. As this is to be fixed soon, I will stamp it to unblock the issue.

**Review by SanabriaRusso — APPROVED** [review]

**Review by SanabriaRusso — CHANGES_REQUESTED** [review]
Move the patch to git as to be self-contained.

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

**Comment by emberian** [comment]
!ci-build-me

File: buildkite/src/Constants/DebianVersions.dhall
Comment (dkijania) [inline buildkite/src/Constants/DebianVersions.dhall:50]: I praise you for finding a way to provide logic for overriding default toolchain runner. For now this change is good, but if we will abandon berkeley_migration profile and use standard like for other, how then we determine that we need to use caqtiRunner?

I wonder if we just won't have a logic on this level.. Instead i would see a global approach on a job level which handle such situation more explicitly, Toolchain runner cannot be determined purely on profile and debian version. What if we will introduce another dhall constant like ToolchainRunner which will encapsulate entire logic of selecting runner? It can have of course standard toolchain but we could also introduce some override mechanism like utilizing some enum:

let ToolchainSelectionMode = < Default | Override: string>

let toolchainRunner = \(debVersion: DebVersion) -> \(mode:  ToolchainSelectionMode)

...

Then we can move selecting toolchain on job level and add some comment so we won't pollute standard toolchains selection and we could explictly write something like:
```

Pipeline.build 
    (ArtifactPipelines.pipeline 
        [ Artifacts.Type.ArchiveMigration ]
        DebianVersions.DebVersion.Bullseye 
        (ToolchainSelectionMode.Type.Override "gcr.io.....") 
        Profiles.Type.BerkeleyMigration
        PipelineMode.Type.PullRequest
    )
```
What you think about it?

I know we are under time pressure so i don't push and i can help you with the change or we can just note that we will do it in the future like that.

File: src/app/replayer/replayer.ml
Comment (georgeee) [inline src/app/replayer/replayer.ml:1749]: Forgotten?

File: buildkite/src/Constants/ContainerImages.dhall
Comment (georgeee) [inline buildkite/src/Constants/ContainerImages.dhall:11]: A comment of what's so special about caqti toolchain would be helpful, I think

File: buildkite/scripts/build-release.sh
Comment (emberian) [reply buildkite/scripts/build-release.sh:27]: TIL 🙏🏼

File: src/app/replayer/replayer.ml
Comment (emberian) [reply src/app/replayer/replayer.ml:1749]: not, otherwise no CI for this build :(

Comment (emberian) [reply src/app/replayer/replayer.ml:1749]: will remove before final merge

**Comment by emberian** [comment]
!ci-build-me

**Pull request merged**
